src/NoteWorkspace.ts

```diff
@@ -39,10 +39,10 @@ type Config = {
 export class NoteWorkspace {
   // Defining these as strings now, and then compiling them with accessor methods.
   // This will allow us to potentially expose these as settings.
-  static _rxTagNoAnchors = '\\#[\\w\\-\\_]+'; // used to match tags that appear within lines
-  static _rxTagWithAnchors = '^\\#[\\w\\-\\_]+$'; // used to match entire words
+  static _rxTagNoAnchors = '\\#[\\w\\p{L}\\-\\_]+'; // used to match tags that appear within lines
```

Comment:
I am unfamiliar with the `\p{L}` regex special character -- can you link me to the documentation for it? 

Will this start matching characters like `\` and `$` or `?` that may have a special meaning on some filesystems?

Also, I think we should have new test cases that assert the expected behavior for this new patter.

For example, there are already some test cases for the `noteFileNameFromTitle` function handling of unicode chars here:

https://github.com/kortina/vscode-markdown-notes/blob/master/src/test/jest/extension.test.ts#L32

We should add some for this regex.

Reply:
Sure, here's a great document about Unicode regular expressions (https://www.regular-expressions.info/unicode.html). Things about `\p{L}` can be found in section "Unicode Categories". It basically matches "all letters in all the languages".

And, yep, I'd like to add some unit tests about this in the near future.

Reply:
Hi @kortina , I've just added some unit tests for the Chinese characters. Other languages should work in the same way as long as they are also Unicode encoded. Please take a look when you have time.